build_scripts/job/dreambooth/utils/utils.py

```
@@ -0,0 +1,259 @@
+        else:
+            print(f"{msg}({allocated}/{cached})")
+
+
+def cleanup(do_print: bool = False):
+    try:
+        if torch.cuda.is_available():
+            torch.cuda.empty_cache()
+            torch.cuda.ipc_collect()
+        gc.collect()
+    except:
+        print("cleanup exception")
+    if do_print:
+        print("Cleanup completed.")
+
+
+def xformers_check():
+    env_vars_true_values = {"1", "ON", "YES", "TRUE"}
+    env_vars_true_and_auto_values = env_vars_true_values.union({"AUTO"})
+
+    use_tf = os.environ.get("USE_TF", "AUTO").upper()
+    use_torch = os.environ.get("USE_TORCH", "AUTO").upper()
+    if (
+        use_torch in env_vars_true_and_auto_values
+        and use_tf not in env_vars_true_values
+    ):
+        _torch_available = importlib.util.find_spec("torch") is not None
+
+        if _torch_available:
+            try:
+                _torch_version = importlib_metadata.version("torch")
+            except importlib_metadata.PackageNotFoundError:
+                print("No metadatapackage")
+                _torch_available = False
+    else:
+        _torch_available = False
+
+    try:
+        _xformers_version = importlib_metadata.version("xformers")
+        if _torch_available:
+            import torch
+            if version.Version(torch.__version__) < version.Version("1.12"):
+                raise ValueError("PyTorch version must be >= 1.12")
+            if version.Version(_xformers_version) < version.Version("0.0.17.dev"):
+                raise ValueError("Xformers version must be >= 0.0.17.dev")
+        has_xformers = True
+    except Exception as e:
+        # print(f"Exception importing xformers: {e}")
+        has_xformers = False
+
+    return has_xformers
+
+
+def list_optimizer():
+    optimizer_list = ["Torch AdamW"]
+
+    try:
+        if shared.device.type != "mps":
+            from bitsandbytes.optim import AdamW8bit
+            optimizer_list.append("8bit AdamW")
+    except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```

Similar issue at line numbers 129, 135, 141, 147, and 153.